### ISSUE TITLE
Invalid `og:description`

### ISSUE THREAD

**Issue by TheAwiteb** [LemmyNet/lemmy#2479]
Hey, i think `og:description` should not have a html elements in it. for example this site

> ![image](https://user-images.githubusercontent.com/59842932/193706855-9b599074-85cb-4f19-ae19-5a4f004d97f6.png)

it's looks like this

<div dir="rtl">

```
مرحبآ انا <strong>بوت</strong> اخبار السعودية، راح انشر اخر اخبار صحيفة اخبار السعودية <a href=&quot;http://news50.sa&quot;>news50.sa</a> 📰 🇸🇦
```

</div>

i think it's should be looks like this

<div dir="rtl">

```
مرحبآ انا بوت اخبار السعودية، راح انشر اخر اخبار صحيفة اخبار السعودية news50.sa 📰 🇸🇦
```
</div>
---

I'am new in [@LemmyNet](https://github.com/LemmyNet) and i think the issue is here
https://github.com/LemmyNet/lemmy/blob/583ceb25063ce0d2fdd5f30708beb75e566d40a2/crates/api_common/src/request.rs#L71-L75

I want to fix it 🤍

**Comment by Nutomic** [comment]
I cant find anything definitive, but it makes sense that there should be no HTML, otherwise someone could also add a `<script>` tag with malicious code. Fixing it would require HTML escape, [for example with this crate](https://docs.rs/html-escape/latest/html_escape/fn.encode_text.html). But as we only need it in this place, it might be enough to replace those three characters manually, instead of pulling in another dependency.

**Comment by TheAwiteb** [comment]
> I cant find anything definitive, but it makes sense that there should be no HTML, otherwise someone could also add a `<script>` tag with malicious code. Fixing it would require HTML escape, [for example with this crate](https://docs.rs/html-escape/latest/html_escape/fn.encode_text.html). But as we only need it in this place, it might be enough to replace those three characters manually, instead of pulling in another dependency.

I was solved with regex

![1664877216160](https://user-images.githubusercontent.com/59842932/193790672-4b4f7058-6b24-4d4f-b38b-516a63b9273f.jpg)

But i want to know there is a another place to fix it? I will create a PR today

**Comment by dessalines** [comment]
The lemmy-ui front end doesn't render that, but yes its possible to sanitize that on the rust side of things. Make sure to use a rust crate tho to sanitize it like @Nutomic mentioned.

**Comment by TheAwiteb** [comment]
@Nutomic I don't know where the front gets the "og:description" from.

**Comment by TheAwiteb** [comment]
It's not from here, I tried to change it but it's still the same

https://github.com/LemmyNet/lemmy/blob/583ceb25063ce0d2fdd5f30708beb75e566d40a2/crates/api_common/src/request.rs#L71-L75

**Comment by Nutomic** [comment]
Did you test by creating a new post? Because description for existing posts wont get updated later.

**Comment by TheAwiteb** [comment]
> Did you test by creating a new post? Because description for existing posts wont get updated later.

No, the bug in the user profile, this description is the bio of the user.

**Comment by Nutomic** [comment]
Can you post a link to the user profile? Im pretty sure og:description isnt used anywhere else besides the box below post.

![Screenshot_20221020_223827](https://user-images.githubusercontent.com/1044450/197053279-31bf94e4-69eb-4121-81f7-7d472a2c8e45.png)

**Comment by TheAwiteb** [comment]
> Can you post a link to the user profile? Im pretty sure og:description isnt used anywhere else besides the box below post.

![image](https://user-images.githubusercontent.com/59842932/197055471-f5819522-d657-4b9a-a381-959902d7e543.png)

**Comment by TheAwiteb** [comment]
> Can you post a link to the user profile? Im pretty sure og:description isnt used anywhere else besides the box below post.

![image](https://user-images.githubusercontent.com/59842932/197055471-f5819522-d657-4b9a-a381-959902d7e543.png)

**Comment by TheAwiteb** [comment]
This is screenshot, i have create account in [lemmy.ml](https://lemmy.ml/), but is not approval yet, the username is `test_for_2479`

**Comment by Nutomic** [comment]
The bio field just shows what you enter manually under /settings. It definitely doesnt use og:description. Nevertheless, this issue needs to be fixed for post metadata.

**Comment by TheAwiteb** [comment]
![image](https://user-images.githubusercontent.com/59842932/197172682-d5cac3a1-23c3-4aae-9646-6314975bf178.png)

See, i think [telegram](https://telegram.org/) get the description from `og:description` or `description` i don't know, i think we should fix both

**Comment by TheAwiteb** [comment]
Of course the fix is not for telegram only, it's for all app that implement url view

**Comment by Nutomic** [comment]
Wait i though this issue was about Lemmy fetching og:description wrongly from other sites. But its really about Lemmy generating og:description wrong? In that case its done somewhere in lemmy-ui.

**Comment by TheAwiteb** [comment]
> Wait i though this issue was about Lemmy fetching og:description wrongly from other sites. But its really about Lemmy generating og:description wrong? In that case its done somewhere in lemmy-ui.

I think lemmy-ui get it from the server, i don't know, we need to check before close the issue

**Comment by dessalines** [comment]
Okay I think I see what's going on.

lemmy-ui converts the markdown in bios and descriptions to html, so that it can be properly rendered by other sites.

https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/components/common/html-tags.tsx#L40

But this is probably incorrect, as the description field should be plain text. If markdown-it doesn't support that, then I can use something like https://www.npmjs.com/package/html-to-text to strip out the html tags. 

@TheAwiteb don't worry I'll get this one, should only take a minute.
